### PR TITLE
[doc] update booster tutorials

### LINKED ISSUE #3717: [booster] update booster tutorials

### PR DESCRIPTION
**Pull request by jiangmingyan**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

resolved #3717 

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

this work is to upadate booster tutorials, which guide user to Initialize parallellism feathures.

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: docs/source/en/basics/colossalai_booster.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+<p>Plugin is an important component that manages parallel configuration (eg: The gemini plugin encapsulates the gemini acceleration solution). Currently supported plugins are as follows:</p>
+
+***GeminiPlugin:*** <p> This plugin wrapps the Gemini acceleration solution, that ZeRO with chunk-based memory management. </p>
+
+***TorchDDPPlugin:*** <p>This plugin wrapps the DDP acceleration solution, it implements data parallelism at the module level which can run across multiple machines. </p>
+
+***LowLevelZeroPlugin:*** <p>This plugin wraps the 1/2 stage of Zero Redundancy Optimizer. Stage 1 : Shards optimizer states across data parallel workers/GPUs. Stage 2 : Shards optimizer states + gradients across data parallel workers/GPUs.</p>
+
+### API of booster
+Booster.__init__(...):
```
Comment (ver217) [inline docs/source/en/basics/colossalai_booster.md:20]: You can use API reference here like https://github.com/hpcaitech/ColossalAI/blob/b37797ed3d3d6af294a095397b4bc135264b8c6a/docs/source/en/features/nvme_offload.md?plain=1#L258

File: docs/source/en/basics/colossalai_booster.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
+# colossal-ai booster
```
Comment (ver217) [inline docs/source/en/basics/colossalai_booster.md:1]: As discussed in our plan, can we rename this file to "booster_api.md"

File: docs/source/en/basics/colossalai_booster.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+**Prerequisite:**
+- [Distributed Training](../concepts/distributed_training.md)
+- [Colossal-AI Overview](../concepts/colossalai_overview.md)
+
+## Introduction
+In our new design, `colossalai.booster` replaces the role of `colossalai.initialize` to inject features into your training components (e.g. model, optimizer, dataloader) seamlessly. With these new APIs, user can integrate their model with our parallelism features more friendly. Also calling `colossalai.booster` is the standard procedure before you run into your training loops. In the sections below, I will cover how `colossalai.booster` works and what we should take note of.
+
+### Plugin
+<p>Plugin is an important component that manages parallel configuration (eg: The gemini plugin encapsulates the gemini acceleration solution). Currently supported plugins are as follows:</p>
```
Comment (ver217) [inline docs/source/en/basics/colossalai_booster.md:11]: Is it essential to mix html and markdown syntax here?

File: docs/source/en/basics/colossalai_booster.md (hunk 2 above)
Comment (ver217) [reply docs/source/en/basics/colossalai_booster.md:1]: Please add this file in `docs/sidebars.json`

File: docs/source/en/basics/booster_api.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+{{ autodoc:colossalai.booster.Booster.save_optimizer }}
+
+{{ autodoc:colossalai.booster.Booster.load_optimizer }}
+
+{{ autodoc:colossalai.booster.Booster.save_lr_scheduler }}
+
+{{ autodoc:colossalai.booster.Booster.load_lr_scheduler }}
+
+## usage
```
Comment (ver217) [inline docs/source/en/basics/booster_api.md:42]: Should we use `Usage`?

File: docs/source/en/basics/booster_api.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
+# Booster API
+
```
Comment (ver217) [inline docs/source/en/basics/booster_api.md:2]: Add author info here

File: docs/source/en/basics/booster_api.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    scheduler.step()
+
+    save_path = "./model"
+    booster.save_model(model, save_path, True, True, "", 10, use_safetensors=use_safetensors)
+
+    new_model = resnet18()
+    booster.load_model(new_model, save_path)
+```
+
+if you want to run a example, [click here](../../../../examples/tutorial/new_api/cifar_resnet/README.md)
```
Comment (ver217) [inline docs/source/en/basics/booster_api.md:82]: Example link can be on the top of this doc, like https://github.com/hpcaitech/ColossalAI/blob/b9a8dff7e5a30b80a60c8dbcea4ed20c870afa63/docs/source/en/features/zero_with_chunk.md#L10

File: docs/source/en/basics/booster_api.md (hunk 5 above)
Comment (ver217) [reply docs/source/en/basics/booster_api.md:3]: https://github.com/hpcaitech/ColossalAI/blob/b9a8dff7e5a30b80a60c8dbcea4ed20c870afa63/docs/source/en/basics/colotensor_concept.md#L3

File: docs/source/en/basics/booster_api.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    scheduler.step()
+
+    save_path = "./model"
+    booster.save_model(model, save_path, True, True, "", 10, use_safetensors=use_safetensors)
+
+    new_model = resnet18()
+    booster.load_model(new_model, save_path)
+```
+
+[more design detailers](https://github.com/hpcaitech/ColossalAI/discussions/3046)
```
Comment (ver217) [inline docs/source/en/basics/booster_api.md:86]: Should it be `details` instead of `detailers`?

File: docs/source/en/basics/booster_api.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+***GeminiPlugin:*** This plugin wrapps the Gemini acceleration solution, that ZeRO with chunk-based memory management.
+
+***TorchDDPPlugin:*** This plugin wrapps the DDP acceleration solution, it implements data parallelism at the module level which can run across multiple machines.
+
+***LowLevelZeroPlugin:*** This plugin wraps the 1/2 stage of Zero Redundancy Optimizer. Stage 1 : Shards optimizer states across data parallel workers/GPUs. Stage 2 : Shards optimizer states + gradients across data parallel workers/GPUs.
+
+### API of booster
+
+
+{{ autodoc:colossalai.booster.Booster.__init__ }}
```
Comment (ver217) [inline docs/source/en/basics/booster_api.md:26]: Docstring of a class can be below the `class` key word, like:
```python
class Person:
"""A class to represent a person.
"""
    pass
```
Use `colossalai.booster.Booster` here to get (maybe) better compatibility.

File: docs/source/zh-Hans/basics/booster_api.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+- [Colossal-AI 总览](../concepts/colossalai_overview.md)
+
+**示例代码**
+- [使用booster训练](https://github.com/hpcaitech/ColossalAI/blob/main/examples/tutorial/new_api/cifar_resnet/README.md)
+
+## 简介
+在我们的新设计中， `colossalai.booster` 代替 `colossalai.initialize` 将特征(例如，模型、优化器、数据加载器）无缝注入您的训练组件中。 使用booster API, 您可以更友好地将我们的并行策略整合到待训练模型中. 调用 `colossalai.booster` 是您进入训练循环前的基本操作。
+在下面的章节中，我们将介绍 `colossalai.booster` 是如何工作的以及使用时我们要注意的细节。
+
+### Plugin
```
Comment (ver217) [inline docs/source/zh-Hans/basics/booster_api.md:15]: Please unify the translation. I notice you use "插件" below, but you use "Plugin" here. What about always use "插件"?

**Review by ver217 — APPROVED** [review]
